### PR TITLE
Let PKI tidy associate revoked certs with their issuers

### PR DESCRIPTION
**Pull request by cipherboy**
This updates PKI's tidy operation to optionally associate certificates with their issuers.

As noted in a previous PR about OCSP, this allows OCSP to behave faster on existing revocation entries (without this mapping) if CRL building is disabled.

---

- [x] Tests
- [x] Docs
- [x] Changelog entry

### CODE REVIEW

File: builtin/logical/pki/crl_test.go [hunk 1]
```diff
@@ -1006,6 +1006,145 @@ func TestAutoRebuild(t *testing.T) {
...
+		require.NoError(t, err)
+		require.NotNil(t, resp)
+		require.NotNil(t, resp.Data)
+		require.NotEmpty(t, resp.Data["state"])
+		state := resp.Data["state"].(string)
+
+		if state == "Finished" {
+			break
+		}
+		if state != "Error" {
```
Comment (stevendpclark) [inline builtin/logical/pki/crl_test.go:1084]: Should this not be "if state == "Error" here and below? If not shouldn't we be continuing on a Running state?

**Review by stevendpclark — APPROVED** [review]
Looks good, one comment about a test that seems a little off to me.

File: builtin/logical/pki/crl_test.go (hunk 1 above)
Comment (cipherboy) [reply builtin/logical/pki/crl_test.go:1084]: Ah yeah, that's right. Oops. :-)

**Review by stevendpclark — APPROVED** [review]
👍

**Pull request merged**